File: app/src/main/java/com/example/justsomeapplication/di/main/fragment/MainFragmentModule.java
package com.example.justsomeapplication.di.main.fragment;

import com.example.justsomeapplication.ui.main.fragment.MainFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface MainFragmentModule {

    @MainFragmentScope
    @ContributesAndroidInjector
    MainFragment fragment();
}

File: app/src/main/java/com/example/justsomeapplication/di/main/fragment/MainFragmentModule.kt
package com.example.justsomeapplication.di.main.fragment

import com.example.justsomeapplication.ui.main.fragment.MainFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface MainFragmentModule {
    @MainFragmentScope
    @ContributesAndroidInjector
    fun fragment(): MainFragment?
}